### PR TITLE
[NFC] Post review of #1230: Asm igemm nhwc native bf16 kernel on gfx90a

### PR DESCRIPTION
**Pull request by shaojiewang**
Fix post review comment of #1230.
Note: Post review of #1227 will be fixed within #1247 by @carlushuang.

----
_Clarification by @atamazov:_ This should resolve bullets (2) and (3) from #1241. No Functional Changes.

### CODE REVIEW

**Comment by junliume** [comment]
> fix post review comment of #1227 and #1230

@shaojiewang Thanks for the quick actions! Please check the tasks in:
https://github.com/ROCmSoftwarePlatform/MIOpen/issues/1241

and if all tasks are completed in this PR, please mark it ready for review :)

**Comment by shaojiewang** [comment]
> > fix post review comment of #1227 and #1230
> 
> @shaojiewang Thanks for the quick actions! Please check the tasks in: #1241
> 
> and if all tasks are completed in this PR, please mark it ready for review :)

Yes, sure.

File: src/include/miopen/solver.hpp [hunk 1]
```diff
@@ -3119,6 +3119,7 @@ struct PerformanceConfigAsmImplicitGemmGTCWrwXdlopsNHWC : PerformanceConfigAsmIm
     bool IsValidValue() const;
     bool IsValid(const ConvolutionContext& ctx) const;
     size_t ComputeKernelOccupancy() const;
+    void SetParamsForKSplit(const ConvolutionContext& ctx, const size_t& occupancy);
```
Comment (atamazov) [inline src/include/miopen/solver.hpp:3122]:
```suggestion
    private:
    void SetParamsForKSplit(const ConvolutionContext& ctx, const size_t& occupancy);
```
basically, `ComputeKernelOccupancy()` should be private as well.

File: src/solver/conv_asm_implicit_gemm_gtc_wrw_nhwc.cpp [hunk 2]
```diff
@@ -594,17 +609,18 @@ void PerformanceConfigAsmImplicitGemmGTCWrwXdlopsNHWC::HeuristicInit(const Convo
...
+            //{
+            //    if(tensor_b_thread_lengths[3] == 1 ||
+            //       miopen::IsDisabled(MIOPEN_DEBUG_CONV_IMPLICIT_GEMM_ASM_PK_ATOMIC_ADD_FP16{}))
+            //        vector_store = 1;
+            //}
+            //else if(ctx.IsBfp16() && tensor_b_thread_lengths[3] == 1)
+            //{
+            //    vector_store = 1;
+            //}
+            //gemm_k_global_split = occupancy;
```
Comment (atamazov) [inline src/solver/conv_asm_implicit_gemm_gtc_wrw_nhwc.cpp]: Please remove

File: src/solver/conv_asm_implicit_gemm_gtc_wrw_nhwc.cpp [hunk 3]
```diff
@@ -635,17 +651,18 @@ void PerformanceConfigAsmImplicitGemmGTCWrwXdlopsNHWC::HeuristicInit(const Convo
...
+                        // {
+                        //     if(tensor_b_thread_lengths[3] == 1 ||
+                        //        miopen::IsDisabled(MIOPEN_DEBUG_CONV_IMPLICIT_GEMM_ASM_PK_ATOMIC_ADD_FP16{}))
+                        //         vector_store = 1;
+                        // }
+                        // else if(ctx.IsBfp16() && tensor_b_thread_lengths[3] == 1)
+                        // {
+                        //     vector_store = 1;
+                        // }
+                        // gemm_k_global_split = occupancy;
```
Comment (atamazov) [inline src/solver/conv_asm_implicit_gemm_gtc_wrw_nhwc.cpp]: ditto

**Review by atamazov — DISMISSED** [review]

File: src/solver/conv_asm_implicit_gemm_gtc_wrw_nhwc.cpp (hunk 2 above)
Comment (shaojiewang) [reply src/solver/conv_asm_implicit_gemm_gtc_wrw_nhwc.cpp]: removed

File: src/solver/conv_asm_implicit_gemm_gtc_wrw_nhwc.cpp (hunk 3 above)
Comment (shaojiewang) [reply src/solver/conv_asm_implicit_gemm_gtc_wrw_nhwc.cpp]: removed

**Review by junliume — APPROVED** [review]

**Review by carlushuang — APPROVED** [review]
LGTM

**Comment by shaojiewang** [comment]
conflicts resolved

**Review by carlushuang — APPROVED** [review]
LGTM

**Review by junliume — APPROVED** [review]

**Comment by junliume** [comment]
@atamazov  gentle ping for review input on this one :)

File: src/solver/conv_asm_implicit_gemm_gtc_wrw_nhwc.cpp [hunk 4]
```diff
@@ -725,7 +720,7 @@ bool PerformanceConfigAsmImplicitGemmGTCWrwXdlopsNHWC::IsValid(const Convolution
     const auto dilation_w = ctx.kernel_size_w > 1 ? ctx.kernel_dilation_w : 1;
     const auto& pad_h     = ctx.pad_h;
     const auto& pad_w     = ctx.pad_w;
-    const auto& precision = ctx.IsFp16() ? miopenHalf : (ctx.IsBfp16() ? miopenBFloat16 : miopenFloat);
+    const auto precision  = ctx.IsFp16() ? miopenHalf : (ctx.IsBfp16() ? miopenBFloat16 : miopenFloat);
```
Comment (atamazov) [inline src/solver/conv_asm_implicit_gemm_gtc_wrw_nhwc.cpp:723]: [Note] https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1230#discussion_r737886602

File: src/solver/conv_asm_implicit_gemm_gtc_wrw_nhwc.cpp [hunk 5]
```diff
@@ -404,6 +404,21 @@ size_t PerformanceConfigAsmImplicitGemmGTCWrwXdlopsNHWC::ComputeKernelOccupancy(
     return occupancy;
 }
 
+void PerformanceConfigAsmImplicitGemmGTCWrwXdlopsNHWC::SetParamsForKSplit(const ConvolutionContext& ctx, const size_t& occupancy)
```
Comment (atamazov) [inline src/solver/conv_asm_implicit_gemm_gtc_wrw_nhwc.cpp:407]: Resolves https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1230#discussion_r737880526

**Review by atamazov — CHANGES_REQUESTED** [review]
- This is not fixed: https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1230#discussion_r737532105

**Comment by shaojiewang** [comment]
> * This is not fixed: [#1230 (comment)](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1230#discussion_r737532105)

let's move to  [#1230 (comment)](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1230#discussion_r737532105)

**Review by atamazov — APPROVED** [review]
LGTM!

**Pull request merged**